[16.0][MIG] shopinvader_restapi: Migration to 16.0

qgroulard opened the pull request:
Based on https://github.com/shopinvader/odoo-shopinvader/pull/1382

qgroulard committed:
[IMP] shopinvader_api_address: test vat format

Make VAT used in tests pass VAT format check

sebastienbeau committed:
Split: Rename module

sebastienbeau committed:
shopinvader_restapi, shopinvader_product_binding: add migration script

qgroulard committed:
[MIG] shopinvader_restapi: Migration to 16.0

qgroulard committed:
[REF] Remove base_url dependency

marielejeune commented on shopinvader_restapi/views/shopinvader_menu.xml:
```diff
@@ -0,0 +1,55 @@
+        id="menu_shopinvader_root"
+        name="Shopinvader"
+        sequence="10"
+        web_icon="shopinvader,static/description/icon.png"
```
This path must be changed since `shopinvader_restapi` does not depends on `shopinvader` anymore

marielejeune commented on shopinvader_restapi/services/cart.py:
```diff
@@ -0,0 +1,533 @@
+            "note": {"type": "string"},
+        }
+
+    def _validator_add_item(self):
```
Return validators are missing for `add_item`, `update_item` and `delete_item`. They are now almost required in v16, as suggested in logs:

 WARNING odoo-test odoo.addons.base_rest.components.service: DEPRECATED: You must define an output schema for method add_item in service shopinvader.cart.service

hparfr commented on shopinvader_restapi/models/shopinvader_backend.py:
```diff
@@ -0,0 +1,369 @@
+    sequence_id = fields.Many2one(
+        "ir.sequence", "Sequence", help="Naming policy for orders and carts"
+    )
+    lang_ids = fields.Many2many("res.lang", string="Lang", required=True)
```
```suggestion
    lang_ids = fields.Many2many("res.lang", string="Lang")
```

There is no need anymore to force to specify the lang.

hparfr commented on shopinvader_restapi/models/shopinvader_backend.py:
```diff
@@ -0,0 +1,369 @@
+        string="Available partner titles",
+        default=lambda self: self._default_partner_title_ids(),
+    )
+    partner_industry_ids = fields.Many2many(
```
can we move this one to a dedicated module ?

hparfr commented on shopinvader_restapi/models/shopinvader_backend.py:
```diff
@@ -0,0 +1,369 @@
+    def _extract_configuration(self):
+        return {}
+
+    def _bind_langs(self, lang_ids):
+        self.ensure_one()
+        self.env["shopinvader.variant.binding.wizard"].bind_langs(self, lang_ids)
+        self.env["shopinvader.category.binding.wizard"].bind_langs(self, lang_ids)
+
+    def _unbind_langs(self, lang_ids):
+        self.ensure_one()
+        self.env["shopinvader.variant.unbinding.wizard"].unbind_langs(self, lang_ids)
+        self.env["shopinvader.category.unbinding.wizard"].unbind_langs(self, lang_ids)
```
```suggestion

```

hparfr commented on shopinvader_restapi/models/shopinvader_backend.py:
```diff
@@ -0,0 +1,369 @@
+                record._bind_langs(list(added_lang_ids))
+            removed_lang_ids = old_lang_ids - actual_lang_ids
+            if removed_lang_ids:
+                record._unbind_langs(list(removed_lang_ids))
```
```suggestion
```

hparfr commented:
there is no `shopinvader_backend.xml`

~~(note: I'm not conviced shopinvader_backend should be in this module, but it should be somewhere :) ) ~~
it will be probably here in the end

marielejeune commented on shopinvader_restapi/tests/common.py:
```diff
@@ -0,0 +1,339 @@
+        # and allow to get rid of this hack.
+        # For full story see
+        # https://github.com/shopinvader/odoo-shopinvader/pull/783
+        if not hasattr(self.env["shopinvader.variant"], "recompute_json"):
```
This condition has to be adapted as `shopinvader.variant` model doesn't exist anymore.

Commit:
review change

Commit:
split shopinvader backend view

Commit:
move filter_ids

Commit:
adapt test class method for product

Commit:
fix

qgroulard committed:
[REF] Remove lang binding

sbidoul commented on shopinvader_restapi/models/sale.py:
```diff
@@ -0,0 +1,133 @@
+    _name = "sale.order"
+    _inherit = ["sale.order", "track.external.mixin"]
+
+    typology = fields.Selection([("sale", "Sale"), ("cart", "Cart")], default="sale")
```
This typology field is now defined in the `sale_cart` module, so should we depend on it to enable co-existence of the old and new cart apis? @lmignon @qgroulard @marielejeune .

marielejeune replied:
Agreed. Then adapt the code in this module as `action_confirm_cart` and `action_confirm` are implemented in `sale_cart` too.

qgroulard committed:
[REF] Sale order typology moved to sale_cart

lmignon committed:
[IMP] shopinvader_restapi: Set development status to 'Alpha'

The status will be changed as soon as a first implementation will be deployed in production

lmignon commented:
/ocabot merge nobump
